src/webapi/document.rs

```diff
@@ -34,6 +35,17 @@ pub fn document() -> Document {
 }
 
 impl Document {
+    /// In an HTML document, the Document.createDocumentFragment() method creates a
+    /// new empty DocumentFragment.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment)
+    // https://dom.spec.whatwg.org/#ref-for-dom-document-createdocumentfragment
+    pub fn create_document_fragment( &self ) -> Result< DocumentFragment, TODO > {
```

Comment:
Why does this return a `Result`? Can `createDocumentFragment` actually throw?

Reply:
Sorry I'm new to this and I don't quite understand what Result does to be honest....but I guess DocumentFragment has the basic functions of Element so I copied and modified create_element function. Perhaps there's a better way to write it?

But I've tested it in my project and the DocumentFragment did actually created. And it does append_child and be appended too. Here's an example:

Live: https://togekk.github.io/stdweb-docfrag/
Souce Code: https://github.com/togekk/rust-mvc/blob/documentfragment/src/main.rs

Reply:
@togekk Unlike most other languages, Rust doesn't have exceptions. So the normal way of handling errors is to use `Result`, which makes it clear which things can error and which things cannot error.

I just now looked at the spec, it seems that `createDocumentFragment` never throws errors, so this should return `DocumentFragment`, not `Result<DocumentFragment, TODO>`

Reply:
Also, please do ask if you do not understand anything, we're happy to help!

Reply:
Thanks for explaining! So should I commit again or...?